### PR TITLE
graphviz: 2.42.2 -> 2.47.2

### PR DESCRIPTION
**Pull request by Mindavi**
###### Motivation for this change

Version bump. I didn't know graphviz was depended on by so many packages, so this might be quite impactful.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
    - [x] `dot`: Tested making a simple diagram
    - [ ] Others: not tested
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/graphics/graphviz/base.nix [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 { lib, stdenv, fetchFromGitLab, autoreconfHook, pkg-config, cairo, expat, flex
 , fontconfig, gd, gettext, gts, libdevil, libjpeg, libpng, libtool, pango
-, bison, fetchpatch, xorg ? null, ApplicationServices }:
+, bison, fetchpatch, xorg ? null, ApplicationServices, python3 }:
```
Comment (SuperSandro2000) [inline pkgs/tools/graphics/graphviz/base.nix:5]:
```suggestion
, bison, fetchpatch, xorg, ApplicationServices, python3 }:
```

**Comment by dotlambda** [comment]
2.47.2 is out

**Comment by Mindavi** [comment]
Updated to 2.47.2 and tested if dot still worked again. Seems to be good.

**Comment by dotlambda** [comment]
@ofborg build graphviz graphviz_2_32

**Comment by dotlambda** [comment]
@bjornfor @mogria Please check whether PlantUML works with this version of Graphviz.

**Comment by Mindavi** [comment]
I tried the example linked in the FAQ (https://forum.plantuml.net/?qa=4531/possible-bug-with-relation-in-rectangle-screenshot) and it seems to render good, so I think this version should be fine for using with plantuml.

The issue says 'Note that versions 2.39 to 2.42 do not work very well with PlantUML.', and also that 2.44 is tested.

Below are the rendered images based on the plantuml code from the FAQ. They look good.

example_001
example_002
example_003

**Comment by dotlambda** [comment]
@Mindavi Can you squash the commits concerning graphviz into one?

**Comment by Mindavi** [comment]
Yes, done :)

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**